MapboxCoreNavigation/RouteController.swift

```diff
@@ -118,8 +115,34 @@ open class RouteController: NSObject {
         }
     }
     
+    /**
+     IOS 8601 timestamp of when the `RouteController` was initialized.
+     */
+    public let startTimestamp: String = Date().ISO8601
+    
+    /**
+     Unique idenitifier for the current `RouteController` session.
+     */
+    public let sessionIdentifier = UUID()
+    
+    var lastReRouteLocation: CLLocation?
+
+    var routeTask: URLSessionDataTask?
+    
+    public var userLocationForRerouteEvent: [CLLocation] = []
+    var rerouteIndexLocationBefore = 0
+    var shouldPushRerouteEventEventually = false
+    var rerouteLocationArraySize = 20
+    var countdownToPushEvent = 20
+    var distanceCompleted: CLLocationDistance = 0
+    var distanceRemaining: CLLocationDistance?
+    var durationRemaining: TimeInterval?
```

Comment:
Should we rename these to `lastRerouteDistanceRemaining` or something to clarify that these properties are stored state from the last reroute? Or should we put these all into a single struct property for the last reroute state?

Something like
```swift
struct LastRerouteState {
    var distanceRemaining: CLLocationDistance
    var durationRemaining: CLLocationDistance
    ...etc...
}